01-front-end-basics/templating-handlebars/27-28/readme.md

```diff
@@ -19,110 +21,91 @@
 
-  ```js
-  // compile handlebars template
-  var source = $('#tracks-template').html();
-  var template = Handlebars.compile(source);
-  ```
 
-5. In the success handler for your AJAX call, pass the data that comes back from the API into your template function. The output of the template function is HTML that contains the data from the API. The last step is to append the HTML to the view.
+**Set up a factory line**
 
-  ```js
+4) Using `#each` in the template lets Handlebars know we'll want to repeat the structure inside the `#each` for every element of the array. 
 
-  function renderSpotifyData(data) {
-    // track results are in an array called `items`
-    // which is nested in the `tracks` object
-    var trackResults = data.tracks.items;
-    console.log(trackResults);
+### What is happening in `app.js`?
 
-    // ...
+**Select something to mold**
 
-    // pass in data to render in the template
-    var trackHtml = template({ tracks: trackResults });
+1) Using jQuery we select our `students-list-template` script element.
 
-    // append html to the view
-    $results.append(trackHtml);
-  }
-  ```
+**Make the mold**
+
+2) We give the element's inner html to Handlebars' `compile` function. Handlebars generates a `template` function for us (our html mold, if you wish).
+
+**Pour the plastic into the mold**
+
+3) The template from our html expects an array of students.  We send the template `data27.students` to use as `students` first. Then we use the template function again with `data28.students` as `students`.
+
+**Open our mold to find a casting!**
+
+4) Each time we call it, the template function returns a string of html that it has generated by mashing the data into the template.
+
+**Put the casting on display**
+
+5) We find select the element where we want the new HTML to appear and append the generated html strings to the page using jQuery.
+
+
+## General Handlebars Setup Steps
+
+1. Add the Handlebars CDN to your `index.html` (remember you can go to <a href="https://cdnjs.com" target="_blank">cdnjs</a> to search for CDNs). Make sure to require Handlebars before your custom script file.
+
+2. Next create an element in your `index.html` where you will append the data from your template. Give this element an `id` (e.g. `results`) so you can select it very specifically with jQuery.
+
+3. Create the template inside your `results` element. Give your template an `id` (e.g. `students-list-template`) so you can select it with jQuery. This template acts as an HTML "mad lib" which you'll fill with data. The `{{}}` parameters will be replaced by the data that comes back from your API call.
+
+4. Compile your template in `js/app.js`. Handlebars has a function called `Handlebars.compile(source)`. The source it takes in is the html from inside your template script. Handlebars' `compile` function returns a function, which we'll save to a variable called `template`. This isn't a special varible name; it just reminds us what the function does.
```

Comment:
Maybe just say `app.js` since I think we're going to use scripts but that could change.  See https://github.com/SF-WDI-LABS/html_strings/issues/2